app/controllers/subscriptions_controller.rb

```diff
@@ -26,15 +28,40 @@ def create
     render :new
   end
 
+  def edit
+    # subscription = Subscription.new(search_criteria: search_criteria_params.to_json)
+    @subscription = SubscriptionPresenter.new(@subscription)
+    Auditor::Audit.new(nil, 'subscription.alert.new', current_session_id).log_without_association
+  end
+
+  def edit_search_criteria
+    @jobs_search_form = VacancyAlgoliaSearchForm.new({})
+  end
+
+  def update
+    if subscription.update(subscription_params)
+      Auditor::Audit.new(subscription, "subscription.#{subscription.frequency}_alert.update", current_session_id).log
+      AuditSubscriptionCreationJob.perform_later(@subscription.to_row)
+      SubscriptionMailer.confirmation(subscription.id).deliver_later
+      render :confirm
+    end
+  end
+
   def unsubscribe
-    token = ParameterSanitiser.call(params).require(:subscription_id)
-    @subscription = Subscription.find_and_verify_by_token(token)
     Auditor::Audit.new(@subscription, "subscription.#{@subscription.frequency}_alert.delete", current_session_id).log
     @subscription.delete
   end
 
 private
 
+  def set_subscription
+    @subscription = Subscription.find_and_verify_by_token(token)
+  end
+
+  def token
+    ParameterSanitiser.call(params).require(:id)
```

Comment:
Generic quesiton: why do we only sometimes use a parameter sanitiser?

Reply:
I believe we take extra care because you can create an alert with an email that is from someone else